frontend/src/components/module-viewer/content/blocks/column/ColumnContent.tsx

```diff
@@ -0,0 +1,145 @@
+import { Box, Center, Flex, useDisclosure, VStack } from "@chakra-ui/react";
+import React, { useContext, useState } from "react";
+import { Droppable } from "react-beautiful-dnd";
+
+import {
+  ColumnBlockParam,
+  ColumnBlockState,
+  ContentBlockState,
+} from "../../../../../types/ContentBlockTypes";
+import DeleteModal from "../../../../common/DeleteModal";
+import EditContentOptionsMenu from "../../../EditContentOptionsMenu";
+
+import EditorContext from "../../../../../contexts/ModuleEditorContext";
+import RenderComponents from "../../helpers/ContentBlockTable";
+
+const [CONTENT_BLOCKS, EDIT_MODALS] = RenderComponents();
+
+const ColumnContent = ({
+  block,
+  columnID,
+  index,
+  side,
+  editable = true,
+}: {
+  block: ContentBlockState | null;
+  columnID: string;
+  index: number;
+  side: ColumnBlockParam;
+  editable?: boolean;
+}): React.ReactElement => {
+  const [isHovered, setIsHovered] = useState(false);
+  const {
+    isOpen: isEditModalOpen,
+    onOpen: onEditModalOpen,
+    onClose: onEditModalClose,
+  } = useDisclosure();
+  const {
+    isOpen: isDeleteModalOpen,
+    onOpen: onDeleteModalOpen,
+    onClose: onDeleteModalClose,
+  } = useDisclosure();
+  const context = useContext(EditorContext);
+
+  if (!context) return <></>;
+  const { dispatch } = context;
+
+  const deleteColumn = () => {
+    const content = {
+      [side]: null,
+    } as ColumnBlockState["content"];
+    dispatch({
+      type: "update-block",
+      value: { index, content },
+    });
+    onDeleteModalClose();
+  };
+
+  const editColumn = <T extends ContentBlockState>(
+    sideUpdates: T["content"],
+  ) => {
+    const content = {
+      [side]: {
+        ...block,
+        content: {
+          ...block?.content,
+          ...sideUpdates,
+        },
+      },
+    } as ColumnBlockState["content"];
+    dispatch({
+      type: "update-block",
+      value: { index, content },
+    });
+    onEditModalClose();
+  };
+
+  if (editable && block === null) {
```

Comment:
Optional: converting this and the below if-statement into nested statements might be less error-prone in the future.